### PR TITLE
document and simplify parallelism

### PR DESCRIPTION
**Pull request by gmartinonQM**
# Description

- Complete documentation of `__fit_and_predict_oof_model`
- Reorder outputs of `__fit_and_predict_oof_model`
- rename some variables to increase readability
- simplify the postprocessing code after parallelism

## Type of change

Please check options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Existing unit tests.

# Checklist:

- [x] I have read the [contributing guidelines](https://github.com/simai-ml/MAPIE/blob/master/CONTRIBUTING.rst)
- [x] I have updated the [HISTORY.rst](https://github.com/simai-ml/MAPIE/blob/master/HISTORY.rst) and [AUTHORS.rst](https://github.com/simai-ml/MAPIE/blob/master/AUTHORS.rst) files
- [x] Linting passes successfully : `flake8 . --exclude=doc`
- [x] Typing passes successfully : `mypy mapie examples --strict --config-file mypy.ini`
- [x] Unit tests pass successfully : `pytest -vs --doctest-modules mapie`
- [x] Coverage is 100% : `pytest -vs --doctest-modules --cov-branch --cov=mapie --pyargs mapie`
- [x] Documentation builds successfully : `cd doc; make clean; make html`

### CODE REVIEW

File: mapie/estimators.py [hunk 1]
```diff
@@ -305,28 +310,26 @@ def fit(self, X: ArrayLike, y: ArrayLike) -> MapieRegressor:
...
+                    clone(estimator), X, y, train_index, val_index, k
+                ) for k, (train_index, val_index) in enumerate(cv.split(X))
             )
-            ks, val_folds, predictions, self.estimators_ = zip(*estimators_and_predictions)
-            ks, val_folds, predictions = np.concatenate(ks), np.concatenate(val_folds), np.concatenate(predictions)
-            self.estimators_ = list(self.estimators_)
-            self.k_[val_folds] = ks
-            y_pred = np.empty_like(y, dtype=float)
-            y_pred[val_folds] = predictions
+            self.estimators_, predictions, val_ids, val_indices = map(list, zip(*cv_outputs))
```
Comment (vtaquet) [inline mapie/estimators.py:329]: Bien vu l'utilisation de `map`, je n'ai pas toujours le reflexe de l'utiliser.

File: mapie/estimators.py [hunk 2]
```diff
@@ -305,28 +310,26 @@ def fit(self, X: ArrayLike, y: ArrayLike) -> MapieRegressor:
             The model itself.
         """
         self._check_parameters()
-        estimator = self._check_estimator(self.estimator)
         cv = self._check_cv(self.cv)
+        estimator = self._check_estimator(self.estimator)
         X, y = check_X_y(X, y, force_all_finite=False, dtype=["float64", "object"])
+        y_pred = np.empty_like(y, dtype=float)
```
Comment (vtaquet) [inline mapie/estimators.py:316]: Pourquoi définir `y_pred` ici ? Si la méthode est `naive`, sa définition préalable est inutile.

Comment (gmartinonQM) [reply mapie/estimators.py:316]: ça ne rallonge pas le code ,ça documente ce qu'on attend en terme de `shape`, ça sépare bien la partie "déclarations" et la partie "calcul". Mais c'est vrai que c'est purement esthétique

File: mapie/estimators.py (hunk 1 above)
Comment (gmartinonQM) [reply mapie/estimators.py:329]: 😉

**Pull request merged**